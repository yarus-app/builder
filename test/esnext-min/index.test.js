/* global __dirname */
import { readFile } from 'fs';
import { resolve } from 'path';

import pify from 'pify';
import rimraf from 'rimraf';

import preppy from '../../src/index';

const lazyRead = pify(readFile);
const lazyDelete = pify(rimraf);

jest.setTimeout(20000);

test('Publish Test File via Babel', async () => {
  await lazyDelete(resolve(__dirname, './dist'));

  await preppy({
    root: __dirname,
    quiet: true,
  });

  expect(
    await lazyRead(resolve(__dirname, 'dist/index.cjs.min.js'), 'utf8'),
  ).toMatchSnapshot('cjs');
  expect(
    await lazyRead(resolve(__dirname, 'dist/index.esm.min.js'), 'utf8'),
  ).toMatchSnapshot('esm');
  expect(
    await lazyRead(resolve(__dirname, 'dist/index.umd.min.js'), 'utf8'),
  ).toMatchSnapshot('umd');
});
