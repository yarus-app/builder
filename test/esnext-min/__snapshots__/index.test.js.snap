// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Babel: cjs 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
\\"use strict\\"
require(\\"core-js/modules/es.array.includes\\"),require(\\"core-js/modules/es.array.iterator\\"),require(\\"core-js/modules/es.object.assign\\"),require(\\"core-js/modules/es.object.to-string\\"),require(\\"core-js/modules/es.promise\\"),require(\\"core-js/modules/es.set\\"),require(\\"core-js/modules/es.string.iterator\\"),require(\\"core-js/modules/web.dom-collections.iterator\\")
var e,o=(e=require(\\"@babel/runtime/helpers/inheritsLoose\\"))&&\\"object\\"==typeof e&&\\"default\\"in e?e.default:e,r=require(\\"lodash\\")
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===r.camelCase(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",4===Math.pow(2,2)),new Promise((function(e,o){e(\\"resolved\\")})).then((function(e){console.log(\\"Promise:\\",e)}))
console.log(\\"Constant:\\",123)
var s=[1,2,3]
console.log(\\"Supports Array.includes?:\\",s.includes&&s.includes(2))
var l,n=new Set(s)
console.log(\\"Supports Set:\\",n.add(4)),void 0===l&&(l=!0),console.log(\\"Supports default parameters:\\",l)
console.log(\\"X Value from inner scope:\\",\\"inner\\"),console.log(\\"X Value from outer scope:\\",\\"outer\\")
var t=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",t)
var c=function(){function e(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}return e.prototype.helper=function(){console.log(\\"Called helper\\")},e}()
c()
console.log(\\"Initialized class:\\",new c),new Promise((function(e,o){return Promise.resolve(PromiseHelper()).then((function(r){try{return e(42)}catch(s){return o(s)}}),o)}))
//# sourceMappingURL=index.cjs.min.js.map
"
`;

exports[`Publish Test File via Babel: esm 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
import\\"core-js/modules/es.array.includes\\"
import\\"core-js/modules/es.array.iterator\\"
import\\"core-js/modules/es.object.assign\\"
import\\"core-js/modules/es.object.to-string\\"
import\\"core-js/modules/es.promise\\"
import\\"core-js/modules/es.set\\"
import\\"core-js/modules/es.string.iterator\\"
import\\"core-js/modules/web.dom-collections.iterator\\"
import o from\\"@babel/runtime/helpers/inheritsLoose\\"
import{camelCase as e}from\\"lodash\\"
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===e(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",4===Math.pow(2,2)),new Promise((function(o,e){o(\\"resolved\\")})).then((function(o){console.log(\\"Promise:\\",o)}))
console.log(\\"Constant:\\",123)
var r=[1,2,3]
console.log(\\"Supports Array.includes?:\\",r.includes&&r.includes(2))
var s,l=new Set(r)
console.log(\\"Supports Set:\\",l.add(4)),void 0===s&&(s=!0),console.log(\\"Supports default parameters:\\",s)
console.log(\\"X Value from inner scope:\\",\\"inner\\"),console.log(\\"X Value from outer scope:\\",\\"outer\\")
var t=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",t)
var n=function(){function o(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}return o.prototype.helper=function(){console.log(\\"Called helper\\")},o}()
n()
console.log(\\"Initialized class:\\",new n),new Promise((function(o,e){return Promise.resolve(PromiseHelper()).then((function(r){try{return o(42)}catch(s){return e(s)}}),e)}))
//# sourceMappingURL=index.esm.min.js.map
"
`;

exports[`Publish Test File via Babel: umd 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
!function(e,o){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?o(require(\\"core-js/modules/es.array.includes\\"),require(\\"core-js/modules/es.array.iterator\\"),require(\\"core-js/modules/es.object.assign\\"),require(\\"core-js/modules/es.object.to-string\\"),require(\\"core-js/modules/es.promise\\"),require(\\"core-js/modules/es.set\\"),require(\\"core-js/modules/es.string.iterator\\"),require(\\"core-js/modules/web.dom-collections.iterator\\"),require(\\"lodash\\")):\\"function\\"==typeof define&&define.amd?define([\\"core-js/modules/es.array.includes\\",\\"core-js/modules/es.array.iterator\\",\\"core-js/modules/es.object.assign\\",\\"core-js/modules/es.object.to-string\\",\\"core-js/modules/es.promise\\",\\"core-js/modules/es.set\\",\\"core-js/modules/es.string.iterator\\",\\"core-js/modules/web.dom-collections.iterator\\",\\"lodash\\"],o):o(null,null,null,null,null,null,null,null,(e=e||self).lodash)}(this,(function(e,o,s,r,l,n,t,c,i){\\"use strict\\"
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===i.camelCase(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",4===Math.pow(2,2)),new Promise((function(e,o){e(\\"resolved\\")})).then((function(e){console.log(\\"Promise:\\",e)}))
console.log(\\"Constant:\\",123)
var u=[1,2,3]
console.log(\\"Supports Array.includes?:\\",u.includes&&u.includes(2))
var d,a=new Set(u)
console.log(\\"Supports Set:\\",a.add(4)),void 0===d&&(d=!0),console.log(\\"Supports default parameters:\\",d)
console.log(\\"X Value from inner scope:\\",\\"inner\\"),console.log(\\"X Value from outer scope:\\",\\"outer\\")
var m=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",m)
var g=function(){function e(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}return e.prototype.helper=function(){console.log(\\"Called helper\\")},e}()
g()
console.log(\\"Initialized class:\\",new g),new Promise((function(e,o){return Promise.resolve(PromiseHelper()).then((function(s){try{return e(42)}catch(r){return o(r)}}),o)}))}))
//# sourceMappingURL=index.umd.min.js.map
"
`;
